pyterrier/transformer.py

```diff
@@ -260,7 +250,7 @@ def __and__(self, right : 'Transformer') -> 'Transformer':
     def __mod__(self, right : int) -> 'Transformer':
         assert isinstance(right, int)
         from .ops import RankCutoffTransformer
-        return RankCutoffTransformer(self, right)
+        return self >> RankCutoffTransformer(right)
```

Comment:
this changes the implementation of cutoff, and would change how it was rendered graphically. It also adds a level of indirection?

Reply:
This change adds flexibility to the `RankCutoffTransformer`. Now you can run it over an arbitrary dataframe directly, rather than needing a dummy transformer to run first. E.g.,:

```python
my_frame = pd.DataFrame(...)
# I want only the top 10 results of my_frame. Previously I'd have to:
my_frame_10 = RankCutoffTransformer(pt.Transformer.identity(), 10)(my_frame)
# Now I can:
my_frame_10 = RankCutoffTransformer(10)(my_frame)
```

I'd argue that this makes it better aligned with how most components in PyTerrier work—take input, transform, output.

Can you elaborate a bit on what you mean by indirection? I sorta see it as more direct to treat it as a regular pipeline component, rather than attach it to an (sometimes arbitrary) other transformer. I.e., defining the retrieval procedure as:

 1. Retrive
 2. Re-Rank
 3. Print results (debug)
 4. Take the top 10

Makes more sense to me than a procedure of:

 1. Retrive
 2. Re-Rank
 3. Print results (debug) AND take the top 10, consisting of:
     1. Print results (debug)
     2. Take the top 10